added storm wiki frontend with extra themes

jaigouk opened the pull request:
### related issue 

https://github.com/stanford-oval/storm/issues/64

### changes 

- Dracula soft dark color and other light and dark themes
- Duckduckgo search. and added searxng and SEARXNG_BASE_URL env var as a fallback option
- using ollama and added gpt-4o-mini as a fallback option
- save result files as '*.md'
- add date to to top of the result file
- added arize-phoenix to trace.

### note

because this frontend is using "fallbacks", I thought it would be better to have separate example instead of overwriting the existing demo_light.

some tests files are added and checked with https://github.com/jaigouk/storm_wiki

### screenshots 

### Screenshots

| ![storm-wiki-dark-theme-setting](https://github.com/user-attachments/assets/f84edbdd-eccd-4ff5-9128-476ca1aed48f) | ![storm-wiki-dark-llm-setting](https://github.com/user-attachments/assets/aa7aa342-51a0-45d7-9986-974b3b77eb11) |
| --- | --- |
| ![storm-wiki-dark-search-setting](https://github.com/user-attachments/assets/f55a1b7f-fc85-4e73-8299-a1c20b130328) | ![storm-wiki-dark-article-new](https://github.com/user-attachments/assets/ff928b10-1903-4d5d-ac73-81f1017db570) |
| ![storm-wiki-dark-article-list](https://github.com/user-attachments/assets/da984979-fce5-4bc1-be9e-1d5948568ce8) | ![storm-wiki-dark-create](https://github.com/user-attachments/assets/de3a3695-5ebd-4217-a049-a60eeb84ab33) |

dark and light theme
| ![storm-wiki-dark-article](https://github.com/user-attachments/assets/c7fad8d5-9d7e-4a7c-8052-635aa1a3ba0d) | ![storm-wiki-solarized-article](https://github.com/user-attachments/assets/4b4e413f-9a30-4eb4-918c-09875e97396d) |
| --- | --- |

jaigouk commented:
@shaoyijia  somehow, the previous PR was getting closed automatically. And I wanted to clean up and refine the index page. Now I think it is ready for review

shaoyijia commented:
Hi @jaigouk , sorry I just saw your message. Is this getting closed automatically again or you need more time? Will review it for sure.

jaigouk commented:
@shaoyijia I squashed commits, rebased again main branch and opend a new pr https://github.com/stanford-oval/storm/pull/123

this time I did not check this option
![Screenshot 2024-08-02 at 10 15 02](https://github.com/user-attachments/assets/7f242c53-9d95-4f3f-a212-caba23844fd1)

I hope the PR survives this time.
